test/compilable/issue19724.sh

```diff
@@ -0,0 +1,53 @@
+#! /usr/bin/env bash
+
+if [[ $OS = *"win"* ]]; then exit 0; fi
```

Comment:
This issue is specific to linux, shouldn't it check for that or does it also affect OSX and BSD.

Reply:
Quite right, sorry. Not used to writing shell tests, so this is mostly copied from 15574.